client/my-sites/site-settings/utils.ts

```diff
@@ -13,9 +20,48 @@ type Site = {
 	plan: Plan;
 };
 
-export function hasSiteAnalyticsFeature( site: Site ): boolean | undefined {
+/**
+ * Returns true if the site is on a free plan and a WPCOM site (Simple or Atomic).
+ *
+ * @param site The site to check.
+ * @param state Global state tree.
+ * @param siteId Site ID to check.
+ * @returns bool True if the site is on a free pln and a WPCOM site, false otherwise.
+ */
+export function isFreeWPCOMSite( site: Site, state: string, siteId: string ): boolean | null {
+	if ( ! ( site && site.plan && state ) ) {
+		return false;
+	}
+
+	return isFreePlan( site.plan.product_slug ) && isSiteWPCOM( state, siteId );
+}
+
+export function hasSiteAnalyticsFeature(
+	site: Site,
+	state: string,
+	siteId: string
+): boolean | undefined {
+	if ( state && siteId && isFreeWPCOMSite( site, state, siteId ) ) {
+		return false;
+	}
```

Comment:
Trying to understand why is this needed, it seems that this handles an underlying issue where the plan for Atomic Free sites is set incorrectly as `jetpack_free`, right?

<img width="863" alt="Screen Shot 2021-06-09 at 13 10 06" src="https://user-images.githubusercontent.com/1233880/121344716-5bbc3a00-c924-11eb-9bd5-e37caacc2f67.png">

Would it be possible to fix the root issue in WP.com, so the proper plan is returned? Maybe after doing that we could remove most of these workarounds?

```diff
Index: wp-content/admin-plugins/wpcom-billing/class.wpcom-store-api.php
===================================================================
--- wp-content/admin-plugins/wpcom-billing/class.wpcom-store-api.php    (revision 226908)
+++ wp-content/admin-plugins/wpcom-billing/class.wpcom-store-api.php    (working copy)
@@ -443,7 +443,7 @@
                                $billing_period = 'Biyearly';
                        }
                } else {
-                       $product_id = is_jetpack_site( $blog_id ) ? JETPACK_FREE : WPCOM_FREE_PRODUCT;
+                       $product_id = is_jetpack_site( $blog_id ) && ! is_blog_atomic( get_blog_details( $blog_id ) ) ? JETPACK_FREE : WPCOM_FREE_PRODUCT;
                        $product = WPCOM_Store::get_product( $product_id );
                        $subscription_is_expired = false;
                        $user_is_owner = false;
```

Reply:
> Trying to understand why is this needed, it seems that this handles an underlying issue where the plan for Atomic Free sites is set incorrectly as `jetpack_free`, right?

That's correct, and the issue was discussed here: pd2C1Z-Q-p2

> Would it be possible to fix the root issue in WP.com, so the proper plan is returned? Maybe after doing that we could remove most of these workarounds?

That would be awesome! My understanding is that it's a bigger task to make that change and check that it doesn't break anything and that all of the feature checks continue to work. This marketing settings fix is on our list of issues blocking the free Atomic work, so we're looking for a short-term UI workaround.